src/main/java/appeng/spatial/StorageCellLot.java

```diff
@@ -0,0 +1,85 @@
+package appeng.spatial;
+
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * A lot inside the storage cell world that is assigned to a specific storage cell.
+ */
+public class StorageCellLot {
+
+    private static final int MAX_SIZE = 512;
+
+    /**
+     * Id of the lot.
+     */
+    private final int id;
+
+    /**
+     * The storage size of this dimension. This is dicateted by the pylon structure
+     * size used to perform the first transfer into this dimension. Once it's set,
+     * it cannot be changed anymore.
+     */
+    private final BlockPos size;
+
+    /**
+     * AE2 player id of who primarily owned the network when the storage lot was
+     * allocated.
+     */
+    private final int owner;
+
+    public StorageCellLot(int id, BlockPos size, int owner) {
+        this.id = id;
+        this.size = size;
+        this.owner = owner;
+        if (size.getX() < 1 || size.getY() < 1 || size.getZ() < 1) {
+            throw new IllegalArgumentException("Lot size " + size + " is smaller than minimum size.");
+        }
+        if (size.getX() > MAX_SIZE || size.getY() >= MAX_SIZE || size.getZ() >= MAX_SIZE) {
```

Comment:
This could lead to a couple issues in case some addon would add different cells. Multuple 512x256x512 cells would create an area without any border to the next one.

Assuming `size` is still the current variant with representing the area within the pylons and not the size of the pylon structure. Also technically `getY` would be limited to 254 (256) blocks or less depending on the set build height.
But that might be implicitly prevent as the inner area is always smaller than the pylon area, thus always 2 blocks smaller then the limit.

Reply:
I actually used picked the MAX_SIZE that was already present in 1.12, but never actually checked against. I suppose we should probably rename MAX_SIZE to PLOT_SIZE and add a SAFETY_MARGIN. Then we'd check against PLOT_SIZE - SAFETY_MARGIN ?

Reply:
Safety margin would always be 1 block, or 2 overall. When taking the pylon structure into account, it would provide that by itself and we just have to limit it to `512x256x512`.

Also we'd still need `512` to align it with region files. So more like `PLOT_SIZE = REGION_SIZE - 2`. And then of course ensure it will always have at least 1 wall of matrix frames above and below. And hopefully you cannot glitch through 1 wide walls with an ender pearl.

Reply:
If we can, the safety margin should ofcourse be such that you cannot see your neighbors. 1 block is certainly not sufficient for that.